examples/vln_benchmark.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import sys
+from collections import defaultdict
+
+import habitat
+from habitat.config.default import get_config
+from habitat.core.benchmark import Benchmark
+from habitat.tasks.nav.shortest_path_follower import ShortestPathFollower
+from habitat_baselines.agents.simple_agents import (
+    ForwardOnlyAgent,
+    GoalFollower,
+    RandomAgent,
+    RandomForwardAgent,
+)
+
+
+"""
+RandomAgent:
+    Takes random actions. If within the goal radius, takes action STOP.
+ForwardOnlyAgent:
+    Only takes action MOVE_FORWARD. If within the goal radius, takes action STOP.
+RandomForwardAgent:
+    If within the goal radius, takes action STOP. Else:
+    P(MOVE_FORWARD) = 80%
+    P(TURN_LEFT) = 10%
+    P(TURN_RIGHT) = 10%
+GoalFollower:
+    Tries to take direct route to the goal and takes action STOP when within
+    the goal radius. Turns left or right if |angle_to_goal| > 15 deg.
+ShortestPathAgent:
+    takes the geodesic shortest path to the goal. If within the goal radius,
+    takes action STOP.
+"""
+
+
+def shortest_path_benchmark(env, mode):
+    """
+    Custom benchmark for the shortest path agent because it requires access
+    to habitat_env during each episode.
+    mode either 'geodesic_path' or 'greedy'
+    """
+    goal_radius = env.episodes[0].goals[0].radius
+
+    follower = ShortestPathFollower(env.sim, goal_radius, False)
+    follower.mode = mode
+
+    for episode in range(len(env.episodes)):
+        env.reset()
+
+        if not env.sim.is_navigable(env.current_episode.goals[0].position):
+            print("Goal is not navigable.")
+        while not env.episode_over:
+            best_action = follower.get_next_action(
+                env.current_episode.goals[0].position
+            )
+            observations = env.step(best_action)
+
+    agg_metrics: Dict = defaultdict(float)
+    for m, v in env.get_metrics().items():
+        agg_metrics[m] += v
+    avg_metrics = {k: v / len(env.episodes) for k, v in agg_metrics.items()}
+    return avg_metrics
+
+
+def main():
```

Comment:
Ideally, we would like to have same launcher for different task benchmarks. It should be possible to incorporate task config into `examples/benchmark.py` and make it support VLN as well. But most probably one more registry needed for supported agents/baselines. Let's create follow up issue with merging `benchmark.py` and `vln_benchmark.py`.